src/DataStructures/Tensor/Expressions/TensorExpression.hpp

```diff
@@ -31,12 +31,13 @@
  * UpLo::Lo> ti_a;`, that is, `ti_a` and `ti_B` are place holders for objects of
  * type `TensorIndex<0, UpLo::Lo>` and `TensorIndex<1, UpLo::Up>` respectively.
  */
-template <std::size_t I, UpLo Valence>
+template <std::size_t I, UpLo Valence, bool IsSpacetime = I < 8>
```

Comment:
I think the compiler will scream unless you put `(I < 8)`. I can't remember when it gets angry, but it doesn't like angle brackets in templates.

What if we make the spatial indices start at 100? Is there any thing that breaks with that? This would allow adding e.g. `a0, a1, a2, a3` in the future when someone decides they need a rank 50 tensor...

Reply:
Hmm I think this comment must be on an outdated comment. The compiler didn't complain about this, but doxygen definitely gave a warning. This was fixed in b865aac

I think that changing the spatial index values to start at 100 doesn't break anything. The only constraints to be aware of in the current code, is that contracted indices start at 1000. Anyone looking for a rank 900 spatial tensor? :)

I'll change this!

Reply:
Yea, it was outdated, you're right.

Haha if someone wants a 900 rank tensor they can rewrite the code :P